miflora/miflora_poller.py

```diff
@@ -154,7 +35,8 @@ def __init__(self, mac, cache_timeout=600, retries=3, adapter='hci0'):
         """
         Initialize a Mi Flora Poller for the given MAC address.
         """
-
+        # TODO: the lifecycle of peripheral is a bit strange and might need improvement
+        self.peripheral = None
         self._mac = mac
```

Comment:
The lifecycle for the bluepy.Peripheral is:
1. open a connection
1. trasnfer some data
1. destroy the Perpheral object

So the Peripheral only lives a few seconds until all data is transfered.

I'm not sure what the intended lifecycle of the milora.poller is. Looking at the locking and caching mechanisms it seems to be intended to be accessed by several threads in parallel and will survive for more than 24h (timer for updating the battery status). But I did not see any documentation on the intended usage/behavior.

It does not make any sense to keep the Bluetooth LE connection to the device open for that long as this will drain the battery. So if you want to keep these features of miflora.poller, we somehow need to manage the lifecycle of the bluepy.Peripheral accordingly. The would probably require to change the control flow within the entire class.

when requesting any parameter: 
1. open the connection
1. read ALL parameters and cache them
1. close the connection
1 provide all future parameters from the cache for some time (5 minutes?)

But I'm not sure if this is really necessary. It would be much easier if you just create a new miflora.poller every time you're polling data and the destroy if when you're done extracting the data.